numpyro/hmc_util.py

```diff
@@ -741,20 +742,33 @@ def initialize_model(rng, model, *model_args, init_strategy='uniform', **model_k
         to convert unconstrained HMC samples to constrained values that
         lie within the site's support.
     """
-    model = seed(model, rng)
-    model_trace = trace(model).get_trace(*model_args, **model_kwargs)
-    sample_sites = {k: v for k, v in model_trace.items() if v['type'] == 'sample' and not v['is_observed']}
-    inv_transforms = {k: biject_to(v['fn'].support) for k, v in sample_sites.items()}
-    prior_params = constrain_fn(inv_transforms,
-                                {k: v['value'] for k, v in sample_sites.items()}, invert=True)
-    if init_strategy == 'uniform':
-        init_params = {}
-        for k, v in prior_params.items():
-            rng, = random.split(rng, 1)
-            init_params[k] = random.uniform(rng, shape=np.shape(v), minval=-2, maxval=2)
-    elif init_strategy == 'prior':
-        init_params = prior_params
+    def initialize_one(key, only_params=False):
```

Comment:
nit: would suggest renaming this to `single_chain_init` or `initialize_single_chain` which will be clearer at first glance.